src/main/java/io/cdap/plugin/ToUTF8Action.java

```diff
@@ -179,31 +200,38 @@ public ToUTF8Config(String sourceFilePath, String destFilePath, @Nullable String
     /**
      * Validates the config parameters required for unloading the data.
      */
-    private void validate() throws IllegalArgumentException {
-      try {
-        Charset.forName(charset);
-      } catch (UnsupportedCharsetException e) {
-        throw new IllegalArgumentException("The charset entered is not valid. Please use a value " +
-                     "from https://docs.oracle.com/javase/8/docs/technotes/guides/intl/encoding.doc.html.", e);
+    private void validate(FailureCollector collector) throws ValidationException {
+      if (Strings.isNullOrEmpty(sourceFilePath)) {
+        collector.addFailure("Source file or folder is required.", "Set source file or folder.")
```

Comment:
If the error message and the corrective action are basically the same (like here), it's better to just have the message and leave the corrective action null.